sunpy/image/transform.py

```diff
@@ -59,14 +59,18 @@ def affine_transform(image, rmatrix, order=3, scale=1.0, image_center=None,
     an alternative affine transformation. The two transformations use different
     algorithms and thus do not give identical output.
 
-    When using for `skimage.transform.warp` with order >= 4 or using
-    `scipy.ndimage.affine_transform` at all, "NaN" values will
-    replaced with zero prior to rotation. No attempt is made to retain the NaN
-    values.
+    When using `skimage.transform.warp` with order >= 4 or using
+    `scipy.ndimage.affine_transform` at all, "NaN" values will be replaced with
+    zero prior to rotation. No attempt is made to retain the "NaN" values except when
+    the entire image consists of "NaN" values only and `skimage.transform.warp` is used,
+    where it is passed down as it is.
```

Comment:
This last part ("passed down as it is") isn't actually true.  As currently implemented, you still go through with the `warp()` call on the all-NaN array, which (I think) has the potential of returning an array that is not all NaNs for parts outside the image boundaries.

Reply:
Yes, you're right, but that explanation doesn't refer to the values returned - only the values passed to `warp`, which are NaNs only (for an all-NaN array). It doesn't talk about the return values at all 😅

I've made a commit with all of the other changes, if you still want me to change this let me know and I'll fix it.

Reply:
Ah, I now understand how to read your sentence, but it's still inaccurate/misleading given the current implementation.  Your code's logic branch for an image of all NaNs comes *after* the code to set NaN values to zero if it is needed for a selected transform.  So:

- If the selected transform requires us to replace NaN values, an image full of NaNs will also be entirely replaced.  The NaN values will not be retained even for an image full of NaN values.
- If the selected transform does not require us to replace NaN values, and image full of NaNs is not treated any differently than an image that has only a few NaN values.  The NaN values are not replaced in either image.  Note that the overall paragraph isn't talking about this case anyway.

Reply:
Ahh, got it. I'll remove the last part of the sentence, thanks.